plugins/modules/ndo_tenant_custom_qos_policy.py

```diff
@@ -0,0 +1,474 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Gaspard Micol (@gmicol) <gmicol@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_tenant_custom_qos_policy
+short_description: Manage Custom QoS Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Custom Quality of Service (QoS) Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Gaspard Micol (@gmicol)
+options:
+  template:
+    description:
+    - The name of the tenant template.
+    type: str
+    aliases: [ tenant_template ]
+    required: true
+  name:
+    description:
+    - The name of the Custom QoS Policy.
+    type: str
+    aliases: [ custom_qos_policy ]
+  uuid:
+    description:
+    - The UUID of the Custom QoS Policy.
+    - This parameter is required when the Custom QoS Policy O(name) needs to be updated.
+    aliases: [ custom_qos_policy_uuid ]
+    type: str
+  description:
+    description:
+    - The description of the Custom QoS Policy.
+    - Providing an empty string will remove the O(description="") from the Custom QoS Policy.
+    type: str
+  dscp_mappings:
+    description:
+    - The Differentiated Services Code Point (DSCP) mappings of the Custom QoS Policy.
+    - O(dscp_mappings.dscp_from) and O(dscp_mappings.dscp_to) cannot both be C(unspecified).
```

Comment:
- Both O(dscp_mappings.dscp_from) and O(dscp_mappings.dscp_to) cannot be C(unspecified).